sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -229,22 +228,13 @@ case class ReplaceTableAsSelectStatement(
  * Column data as parsed by ALTER TABLE ... ADD COLUMNS.
  */
 case class QualifiedColType(
-    name: Seq[String],
+    fieldName: FieldName,
```

Comment:
can you summarize what we need to do for it? What I can think of:
1. normalize the field name according to the actual table schema
2. make sure the column name does not exist in the table schema

Reply:
For `AlterTableAddColumns`, we need to resolve the "parent" name if the column being added is a nested one, and make sure the column name doesn't exist.

For `AlterTableReplaceColumns`, it seems that we do not need to check anything with this new change (I removed it in the recent [commit](https://github.com/apache/spark/pull/33200/commits/1eab11f13030c76a500d6d2b46fc8e39a3617d71))

The reason I was using `FieldName` is so that I can check whether `QualifiedColType` is resolved or not so that rule doesn't run if already resolved:
```
private def hasUnresolvedColumns(cols: Seq[QualifiedColType]): Boolean = {
      cols.exists(col => !col.fieldName.resolved || col.position.exists(!_.resolved))
```
, but I agree that using `ResolvedFieldName` is a bit weird since the field name is being "added". Maybe turn `QualifiedColType` into an `Expression`?